### ISSUE TITLE
memory leak in create_from_point_cloud_ball_pivoting

### ISSUE THREAD

**Issue by zjhthu** [intel-isl/Open3D#1360]
**IMPORTANT: Please use the following template to report the bug.**
_____

**Describe the bug**
I may find a memory leak in `o3d.geometry.TriangleMesh.create_from_point_cloud_ball_pivoting`

**To Reproduce**

```
import open3d as o3d
pcd = o3d.io.read_point_cloud('pcd.ply')
avg_dist = 0.06795456469260171
radius = 1.5 * avg_dist
while True:
    mesh = o3d.geometry.TriangleMesh.create_from_point_cloud_ball_pivoting(
       pcd,
       o3d.utility.DoubleVector([radius, radius * 2]))
```

**Environment (please complete the following information):**
 - OS:  Ubuntu 16.04
 - Python version: 3.5
 - Open3D version: 0.8
 - Is this remote workstation?: yes 
 - How did you install Open3D?: pip

**Comment by griegler** [comment]
Can you provide more details? How did you observe a memory leak? Did you check with valgrind, or something similar? I ask because I can not completely rule out a leak, but it seems very unlikely. The code does not use any unmanaged `new`, or `malloc`.

**Comment by zjhthu** [comment]
The memory just grows when the script runs. The ply file used in this scripts is large (83M), you can run the script for a long time (about 10 min)

**Comment by griegler** [comment]
For a single reconstruction it is expected that the memory grows. Also it might be, that the Python garbage collector does not free the mesh in the loop iterations.

**Comment by zjhthu** [comment]
my computer runs out of memory (128G) after running code with this function several hundred times, it should not a normal phenomenon ...

**Comment by griegler** [comment]
can you share the ply file?

**Comment by zjhthu** [comment]
you can try this
https://drive.google.com/file/d/1cL9haqQS0ieeMgkqRE2AIlziORSSrHar/view?usp=sharing

**Comment by griegler** [comment]
Fixed in #1363.

**Comment by zjhthu** [comment]
thx!
